calicoctl/commands/resources.go

```diff
@@ -304,3 +249,51 @@ func executeResourceAction(args map[string]interface{}, client client.Interface,
 
 	return []runtime.Object{resOut}, err
 }
+
+// handleNamespace fills in the namespace information in the resource (if required),
+// and validates the namespace depending on whether or not a namespace should be
+// provided based on the resource kind.
+func handleNamespace(resource runtime.Object, args map[string]interface{}) (resourcemgr.ResourceObject, string, bool, error) {
```

Comment:
Not sure what the string and bool return values are.

The resource input parameter could also be the output parameter, so you don't actually need to include the resourceObject in the results.  In any case, why is the input resource a runtime.Object, but the return value a resurcemgr.ResourceObject?

Reply:
Do you need to return the namespace?  Isn't it included in the resource?

Reply:
It feels like all of the parameters except the error are redundant:
-  The resource on the input parms can also be used as the output, e.g.  `handleNamespace(inout runtime.Object, arg ...)` ... a useful thing I picked up reading the k8s code
-  The namespace can be extracted from the resource
-  The all namespaces can be inferred from the `resource.Namespace == ""`.  From a Get/LIst perspective, just fill in the Name and Namespace from the values in the resource and the blank values will automatically be treated as wildcards.